fvm/environment/derived_data_invalidator.go

```diff
@@ -39,8 +76,8 @@ type ProgramInvalidator struct {
 }
 
 func (invalidator ProgramInvalidator) ShouldInvalidateEntries() bool {
-	// TODO(patrick): invalidate on meter param updates
-	return len(invalidator.ContractUpdateKeys) > 0 ||
+	return invalidator.MeterParamOverridesUpdated ||
+		len(invalidator.ContractUpdateKeys) > 0 ||
```

Comment:
can you explain the case where `invalidator.MeterParamOverridesUpdated` is false, but `len(invalidator.ContractUpdateKeys) > 0 || len(invalidator.FrozenAccounts) > 0` is true?

Also why do we remove `len(invalidator.ContractUpdateKeys) > 0 || len(invalidator.FrozenAccounts) > 0` for `MeterParamOverridesInvalidator` ?

Reply:
any contract update to a regular account for example will trigger this condition.

i've remove this condition from MeterParamOverridesInvalidator because it's wrong (i only added that as a temporary placeholder).  in general, meter param overrides don't change when contract changes.